### PR TITLE
Make the JAX-RS 3.0 file upload in-memory threshold configurable

### PR DESCRIPTION
**Pull request by andymc12**
Currently, the JAX-RS 3.0 client will write to disk when the upload size of the request is >1MB and this is not configurable - this change makes that limit configurable via MP Config.

Configured using MP Config property:
`org.jboss.resteasy.client.jaxrs.engines.fileUploadInMemoryThreshold`

Unit type is MB.  Default is 1.

### CODE REVIEW

**Comment by andymc12** [comment]
#build

**Comment by andymc12** [comment]
Rebased on recent changes and resolved merge conflicts. New #build

File: dev/io.openliberty.org.jboss.resteasy.common/src/org/jboss/resteasy/client/jaxrs/engines/ManualClosingApacheHttpClient43Engine.java [hunk 1]
```diff
@@ -763,14 +779,14 @@ public void close()
    // https://github.com/resteasy/Resteasy/pull/2778
    private static File getTempDir() {
        if (System.getSecurityManager() == null) {
-          final Optional<String> value = ConfigurationFactory.getInstance().getConfiguration().getOptionalValue("java.io.tmpdir", String.class);
+          final Optional<String> value = ResteasyConfigProvider.getConfig().getOptionalValue("java.io.tmpdir", String.class);
```
Comment (WhiteCat22) [inline dev/io.openliberty.org.jboss.resteasy.common/src/org/jboss/resteasy/client/jaxrs/engines/ManualClosingApacheHttpClient43Engine.java:782]: I don't like that this was changed to `ResteasyConfigProvider.getConfig()`. `ResteasyConfigProvider` is deprecated and the RestEasy community gave me this snippet to use.

File: dev/io.openliberty.org.jboss.resteasy.common/src/org/jboss/resteasy/client/jaxrs/engines/ManualClosingApacheHttpClient43Engine.java [hunk 2]
```diff
@@ -123,39 +122,56 @@ public String run() throws Exception
...
+       }
       this.httpClient = httpClient;
       this.httpContextProvider = httpContextProvider;
-      this.allowClosingHttpClient = true;
+      this.allowClosingHttpClient = closeHttpClient;
+      this.defaultProxy = defaultProxy;
+      
+      try {
+          int threshold = Integer.parseInt(
+              ResteasyConfigProvider.getConfig()
```
Comment (WhiteCat22) [inline dev/io.openliberty.org.jboss.resteasy.common/src/org/jboss/resteasy/client/jaxrs/engines/ManualClosingApacheHttpClient43Engine.java:165]: `ResteasyConfigProvider` is deprecated. Is there a reason not to use `ConfigurationFactory.getInstance().getConfiguration()`?

File: dev/io.openliberty.org.jboss.resteasy.common/src/org/jboss/resteasy/client/jaxrs/engines/ManualClosingApacheHttpClient43Engine.java [hunk 3]
```diff
@@ -123,39 +122,56 @@ public String run() throws Exception
...
+                                    .getOptionalValue("org.jboss.resteasy.client.jaxrs.engines.fileUploadInMemoryThreshold", String.class)
```
Comment (WhiteCat22) [inline dev/io.openliberty.org.jboss.resteasy.common/src/org/jboss/resteasy/client/jaxrs/engines/ManualClosingApacheHttpClient43Engine.java:166]: this looks like it needs to be in a do-priv

**Review by WhiteCat22 — CHANGES_REQUESTED** [review]

File: dev/io.openliberty.org.jboss.resteasy.common/src/org/jboss/resteasy/client/jaxrs/engines/ManualClosingApacheHttpClient43Engine.java (hunk 1 above)
Comment (andymc12) [reply dev/io.openliberty.org.jboss.resteasy.common/src/org/jboss/resteasy/client/jaxrs/engines/ManualClosingApacheHttpClient43Engine.java:782]: eventually, we'll need to migrate to the new ConfigurationFactory SPIs, but right now those don't work with Open Liberty due to the differences in our overlays and integration with our config.

File: dev/io.openliberty.org.jboss.resteasy.common/src/org/jboss/resteasy/client/jaxrs/engines/ManualClosingApacheHttpClient43Engine.java (hunk 3 above)
Comment (andymc12) [reply dev/io.openliberty.org.jboss.resteasy.common/src/org/jboss/resteasy/client/jaxrs/engines/ManualClosingApacheHttpClient43Engine.java:166]: No, the config provider has all of the do-privs it needs to read system properties, env properties, etc.

**Review by WhiteCat22 — APPROVED** [review]

**Comment by andymc12** [comment]
#libby

**Pull request merged**